### PR TITLE
feat(utility): migrate requests related source code from TensorBaySDK

### PR DESCRIPTION
**Pull request by Joey9933**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2jn2tnz TensorBay 代码迁移：utility/requests.py](https://app.clickup.com/t/2jn2tnz)

File: graviti/openapi/request.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 from uuid import uuid4
```
Comment (graczhual) [inline graviti/openapi/request.py:9]: It looks like the file should be uniformly named "requests.py"?

Comment (graczhual) [reply graviti/openapi/request.py:9]: "utility/request" -> "client/requests"? @Lee-000

File: graviti/utility/request.py [hunk 2]
```diff
@@ -9,10 +9,10 @@
 from uuid import uuid4
 
 from requests.models import Response
-from tensorbay.utility import config, get_session
 
 from graviti.__version__ import __version__
 from graviti.exception import RESPONSE_ERROR_DISTRIBUTOR, ResponseError
+from graviti.utility import config, get_session
```
Comment (graczhual) [inline graviti/utility/request.py:15]: graviti.utility.requests

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/utility/request.py (hunk 2 above)
Comment (Joey9933) [reply graviti/utility/request.py:15]: I have imported these two objects in utility.__init__.py, is it right?

File: graviti/openapi/request.py (hunk 1 above)
Comment (Lee-000) [reply graviti/openapi/request.py:9]: We could do that in another pr.

File: graviti/utility/requests.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+                return b""
+
+        try:
+            return next(self.response.iter_content(amt))  # type: ignore[no-any-return]
+
+        except StopIteration:
+            return b""
+
+
+class Tqdm(tqdm):  # type: ignore[misc]
```
Comment (Lee-000) [inline graviti/utility/requests.py:225]: `Tqdm` is not used currently. Remove it for now?

Comment (Joey9933) [reply graviti/utility/requests.py:225]: ok

File: graviti/utility/requests.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+import os
+from collections import defaultdict
+from typing import Any, DefaultDict, Optional
+
+import urllib3
+from requests import Session
+from requests.adapters import HTTPAdapter
+from requests.exceptions import RequestException, StreamConsumedError
+from requests.models import PreparedRequest, Response
+from tensorbay.client.log import RequestLogging, ResponseLogging
```
[the remark this answers is not in the document]
Comment (Joey9933) [reply graviti/utility/requests.py]: the log.py is migrated to graviti/client/

File: graviti/client/log.py [hunk 5]
```diff
@@ -0,0 +1,268 @@
+# !/usr/bin/env python3
```
Comment (Lee-000) [inline graviti/client/log.py:1]: Should this file be placed under `graviti/utility` folder? There is a circular import if log is placed here.

**Review by Lee-000 — CHANGES_REQUESTED** [review]

File: graviti/utility/log.py [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+"status_code": {}
+"reason": {}
+"headers": {}
+"content": {}
+"cost_time": {}s
+===================================================================
+"""
+
+
+UPLOAD_DATASET_RESUME_TEMPLATE = """
```
Comment (linjiX) [inline graviti/utility/log.py:35]: The following template is useless in graviti SDK:
- `UPLOAD_DATASET_RESUME_TEMPLATE`
- `UPLOAD_SEGMENT_RESUME_TEMPLATE_SDK`
- `UPLOAD_SEGMENT_RESUME_TEMPLATE_CLI `
- `CACHE_SPACE_WARNING `

File: graviti/utility/__init__.py [hunk 7]
```diff
@@ -8,5 +8,14 @@
 from graviti.utility.file import File
 from graviti.utility.lazy import LazyFactory
 from graviti.utility.request import URL_PATH_PREFIX, open_api_do
+from graviti.utility.requests import config, get_session
 
-__all__ = ["File", "LazyFactory", "open_api_do", "URL_PATH_PREFIX", "AttrDict"]
+__all__ = [
+    "File",
+    "LazyFactory",
+    "open_api_do",
```
Comment (graczhual) [inline graviti/utility/__init__.py:16]: do wee need to add RequestLogging and ResponseLogging here?

Comment (Joey9933) [reply graviti/utility/__init__.py:16]: for now, log.py has been referred only by utility.requests.

**Review by graczhual — APPROVED** [review]

File: graviti/openapi/request.py (hunk 1 above)
Comment (Joey9933) [reply graviti/openapi/request.py:9]: > We could do that in another pr.

ok

File: graviti/utility/request.py (hunk 2 above)
Comment (Joey9933) [reply graviti/utility/request.py]: done

File: graviti/openapi/request.py [hunk 8]
```diff
@@ -9,10 +9,10 @@
...
 
 from requests.models import Response
-from tensorbay.utility import config, get_session
 
 from graviti.__version__ import __version__
 from graviti.exception import RESPONSE_ERROR_DISTRIBUTOR, ResponseError
+from graviti.utility import config, get_session
```
[the remark this answers is not in the document]
Comment (Joey9933) [reply graviti/openapi/request.py]: done

**Review by linjiX — APPROVED** [review]

File: graviti/utility/log.py (hunk 6 above)
Comment (Joey9933) [reply graviti/utility/log.py]: ok, they have been removed.

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**